server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java

```diff
@@ -0,0 +1,141 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ***************************************************************/
+
+package org.apache.james.task.eventsourcing.distributed;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import javax.annotation.PreDestroy;
+
+import org.apache.james.backend.rabbitmq.ReactorRabbitMQChannelPool;
+import org.apache.james.backend.rabbitmq.SimpleConnectionPool;
+import org.apache.james.lifecycle.api.Startable;
+import org.apache.james.server.task.json.JsonTaskSerializer;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskId;
+import org.apache.james.task.TaskManagerWorker;
+import org.apache.james.task.TaskWithId;
+import org.apache.james.task.WorkQueue;
+import org.reactivestreams.Publisher;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableMap;
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Connection;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import reactor.rabbitmq.AcknowledgableDelivery;
+import reactor.rabbitmq.BindingSpecification;
+import reactor.rabbitmq.ConsumeOptions;
+import reactor.rabbitmq.ExchangeSpecification;
+import reactor.rabbitmq.OutboundMessage;
+import reactor.rabbitmq.QueueSpecification;
+import reactor.rabbitmq.RabbitFlux;
+import reactor.rabbitmq.ReceiverOptions;
+import reactor.rabbitmq.Sender;
+import reactor.rabbitmq.SenderOptions;
+
+public class RabbitMQWorkQueue implements WorkQueue, Startable {
+    static final Integer MAX_CHANNELS_NUMBER = 1;
+    static final String EXCHANGE_NAME = "taskManagerWorkQueueExchange";
+    static final String QUEUE_NAME = "taskManagerWorkQueue";
+    static final String ROUTING_KEY = "taskManagerWorkQueueRoutingKey";
+    public static final String TASK_ID = "taskId";
+
+    private final TaskManagerWorker worker;
+    private final Mono<Connection> connectionMono;
+    private final ReactorRabbitMQChannelPool channelPool;
+    private final JsonTaskSerializer taskSerializer;
+    private Sender sender;
+    private RabbitMQExclusiveConsumer receiver;
+
+    public RabbitMQWorkQueue(TaskManagerWorker worker, SimpleConnectionPool simpleConnectionPool, JsonTaskSerializer taskSerializer) {
+        this.worker = worker;
+        this.connectionMono = simpleConnectionPool.getResilientConnection();
+        this.taskSerializer = taskSerializer;
+        this.channelPool = new ReactorRabbitMQChannelPool(connectionMono, MAX_CHANNELS_NUMBER);
+    }
+
+    public void start() {
+        sender = RabbitFlux.createSender(new SenderOptions()
+            .connectionMono(connectionMono)
+            .channelPool(channelPool)
+            .resourceManagementChannelMono(
+                connectionMono.map(Throwing.function(Connection::createChannel)).cache()));
```

Comment:
IMO giving the sender should be a responsibility of the channel pool, that chunk of code is duplicated everywhere

Reply:
It is fixed in #2640